src/nvim/regexp.c

```diff
@@ -4353,7 +4353,7 @@ static int regmatch(
             status = RA_NOMATCH;
           }
 
-          if (status != RA_FAIL && status != RA_NOMATCH) {
+          if (status != RA_NOMATCH) {
```

Comment:
if we "know" it's always true then it should be asserted. then we avoid future regressions.

    assert(status != RA_FAIL);